schema.go

```diff
@@ -411,11 +429,16 @@ func structNodeOf(t reflect.Type) *structNode {
 
 	for i := range fields {
 		field := structField{name: fields[i].Name, index: fields[i].Index}
-		field.Node = makeNodeOf(fields[i].Type, fields[i].Name, []string{
-			fields[i].Tag.Get("parquet"),
-			fields[i].Tag.Get("parquet-key"),
-			fields[i].Tag.Get("parquet-value"),
-		})
+
+		tagKeyToVals := make(map[string]string, len(knownTagKeys)-1)
```

Comment:
What do you think about declaring a struct instead of using a map here? It seems like there's only a finite set of valid fields, a struct type would help make this explicit.